lib/ansible/modules/cloud/amazon/ecr.py

```diff
@@ -0,0 +1,316 @@
+            policy = self.ecr.set_repository_policy(repositoryName=name,
+                                                    policyText=policy_text)
+            self.changed = True
+            return policy
+
+        self.skipped = True
+        if self.get_repository(name) is None:
+            # TODO: if does not exists, fail gracefully
+            raise Error('TODO: handle this case')
+        return
+
+    def delete_repository(self, name):
+        if not self.check_mode:
+            repo = self.ecr.delete_repository(repositoryName=name)
+            self.changed = True
+            return repo
+
+        repo = self.get_repository(name)
+        if repo:
+            self.skipped = True
+            return repo
+        return None
+
+    def delete_repository_policy(self, name):
+        if not self.check_mode:
+            policy = self.ecr.delete_repository_policy(repositoryName=name)
+            self.changed = True
+            return policy
+
+        policy = self.get_repository_policy(name)
+        if policy:
+            self.skipped = True
+            return policy
+        return None
+
+
+def run(ecr, params, verbosity):
+    result = {}
+    try:
+        name = params['name']
+        state = params['state']
+        policy = params['policy']
+        policy_text = params['policy_text']
+        delete_policy = params['delete_policy']
+
+        try:
+            if policy_text is not None:
+                # policy_text given; parse it
+                policy = json.loads(policy_text)
+            elif policy is not None:
+                # policy given; dump it
+                policy_text = json.dumps(policy)
+        except ValueError:
+            result['msg'] = 'Invalid JSON given for policy_text'
+            return False, result
+
+        result['state'] = state
+        result['created'] = False
+
+        repo = ecr.get_repository(name)
+
+        if state == 'present':
+            result['created'] = False
+            if not repo:
+                repo = ecr.create_repository(name)
+                result['changed'] = True
+                result['created'] = True
+            result['repository'] = repo
+
+            if delete_policy:
+                original_policy = ecr.get_repository_policy(name)
+
+                if verbosity >= 2:
+                    result['policy'] = None
+
+                if verbosity >= 3:
+                    result['original_policy'] = original_policy
+
+                if original_policy:
+                    ecr.delete_repository_policy(name)
+                    result['changed'] = True
+
+            elif policy is not None:
+                try:
+                    if verbosity >= 2:
+                        result['policy'] = policy
+                    original_policy = ecr.get_repository_policy(name)
+                    if verbosity >= 3:
+                        result['original_policy'] = original_policy
+
+                    if original_policy != policy:
```

Comment:
Good catch. Looks like the guidelines have been updated to recommend `sort_json_policy_dict()`, which should be sufficient for comparing policy docs.